verification/io/router.gobra

```diff
@@ -73,16 +128,27 @@ pure func insert(buf dict[option[IO_ifs]](set[IO_pkt3]), k option[IO_ifs], v IO_
 		buf[k = newSet]
 }
 
-// TODO: try to remove the existencials here
 ghost
+requires len(m.CurrSeg.Future) > 0
+requires len(newpkt.CurrSeg.Future) > 0
+requires dp.Valid()
 decreases
 pure func (dp DataPlaneSpec) dp3s_forward_ext(m IO_pkt3, newpkt IO_pkt3, nextif IO_ifs) bool {
-	return (exists currseg IO_seg3, traversedseg IO_seg3, fut seq[IO_HF], hf1 IO_HF :: { dp.dp2_forward_ext_guard(dp.asid(), m, nextif, currseg, traversedseg, newpkt, fut, hf1) } dp.dp2_forward_ext_guard(dp.asid(), m, nextif, currseg, traversedseg, newpkt, fut, hf1)) &&
-			(exists a2 IO_as, i2 IO_ifs :: { dp.is_target(dp.asid(), nextif, a2, i2) } dp.is_target(dp.asid(), nextif, a2, i2))
+	return let currseg := m.CurrSeg in
+		let hf1, fut := currseg.Future[0], currseg.Future[1:] in
+		let traversedseg := dp3s_iospec_bio3s_create_guard_traversedseg_inc(currseg) in
+		dp.dp2_forward_ext_guard(dp.asid(), m, nextif, currseg, traversedseg, newpkt, fut, hf1) &&
+		let hf2 := newpkt.CurrSeg.Future[0] in
```

Comment:
do I see it correctly that `hf2` is not used anywhere? Why did we add it? I guess that, if we drop it, we can also drop the precondition `len(newpkt.CurrSeg.Future) > 0` (and drop this precondition from every function that calls this one)

Reply:
Yes `hf2` can be removed.